caniusepython3/__main__.py

```diff
@@ -51,6 +51,22 @@ def projects_from_requirements(requirements):
     return valid_reqs
 
 
+def req_has_file_link(req):
+    url = getattr(req, 'url', '')
+    if url and is_file_link(url):
+        return True
+    link = getattr(req, 'link', '')
+    if link and is_file_link(link):
+        return True
+
+
+def is_file_link(link):
```

Comment:
Having this as a separate function seems unnecessary, especially if you're special-casing in both functions. You might as well just take the logic from here and put it in req_has_file_link() so you don't have to do the hasattr() checks.